src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs

```diff
@@ -110,13 +110,24 @@ private static void ValidateCryptoStream(string expected, string data, ICryptoTr
             byte[] inputBytes = Text.Encoding.ASCII.GetBytes(data);
             byte[] outputBytes = new byte[100];
 
+            // Verify read mode
             using (var ms = new MemoryStream(inputBytes))
             using (var cs = new CryptoStream(ms, transform, CryptoStreamMode.Read))
             {
                 int bytesRead = cs.Read(outputBytes, 0, outputBytes.Length);
                 string outputString = Text.Encoding.ASCII.GetString(outputBytes, 0, bytesRead);
                 Assert.Equal(expected, outputString);
             }
+
+            // Verify write mode
+            using (var ms = new MemoryStream(outputBytes))
+            using (var cs = new CryptoStream(ms, transform, CryptoStreamMode.Write))
+            {
+                cs.Write(inputBytes, 0, inputBytes.Length);
```

Comment:
Write mode may be how the problem was found, and therefore is a valid test case, but since the problem was lower level it would be good to see a test with less machinery.

This code, for instance, fails.  The buffer provided did not have whitespace in the first `inputCount` bytes, and when the whole array is returned it over-reads.

```C#
            using (var transform = new FromBase64Transform())
            {
                byte[] input = { (byte)'E', (byte)' ', (byte)'L', (byte)'L' };
                byte[] expectedOutput = { 0x10, 0x62, 0xCB };
                byte[] output = new byte[expectedOutput.Length];

                // Write 'E'
                int written = transform.TransformBlock(input, 0, 1, output, 0);
                Assert.Equal(0, written);

                // Instead of using input[1], overwrite input[0] with 'G' and read that.
                input[0] = (byte)'G';
                written = transform.TransformBlock(input, 0, 1, output, written);
                Assert.Equal(0, written);

                // Write the "LL" which completes a 4 => 3 and a write.
                written = transform.TransformBlock(input, 2, 2, output, written);
                Assert.Equal(output.Length, written);
                Assert.Equal(expectedOutput, output);
            }
```